crates/data_components/src/s3_vectors/query_provider.rs

```diff
@@ -146,38 +156,137 @@ impl TableProvider for S3VectorsQueryTable {
             .collect())
     }
 
+    #[allow(clippy::too_many_lines)]
```

Comment:
Is there an opportunity for some refactoring to get rid of #[allow(clippy::too_many_lines)]?